.clang-format

```diff
@@ -0,0 +1,219 @@
+AccessModifierOffset: -2
+IndentCaseLabels: false
+EmptyLineBeforeAccessModifier: Always
+ConstructorInitializerIndentWidth: 2
+
+#Always break after an open bracket, if the parameters don’t fit on a single line
+AlignAfterOpenBracket: AlwaysBreak
+
+# Newline escapes should be aligned and located at the leftmost alignment
+AlignEscapedNewlines: Left
+
+# Align after operands across continuation into multiple lines for binary and
+# ternary.
+AlignOperands: Align
+
+# Do not align trailing comments across multiple lines. Note that this only applies
+# to the single-line comment style using "//". Block inlines are always left alone.
+AlignTrailingComments: false
+
+# If the function declaration doesn’t fit on a line, allow putting all parameters of a
+# function declaration onto the next line even if BinPackParameters is OnePerLine
+AllowAllParametersOfDeclarationOnNextLine: false
+
+# Do not allow short enums on a single line.
+AllowShortEnumsOnASingleLine: false
+
+# Only merge functions defined inside a class. Implies empty
+AllowShortFunctionsOnASingleLine: Inline
+
+# Never put short ifs on the same line.
+AllowShortIfStatementsOnASingleLine: false
+
+# If true, "while (true) continue;" can be put on a single line
+AllowShortLoopsOnASingleLine: false
+
+# Always break after the return type of function definitions.
+AlwaysBreakAfterReturnType: AllDefinitions
+
+# Always break after template declaration.
+AlwaysBreakTemplateDeclarations: Yes
+
+# If false, a function declaration’s or function definition’s parameters
+# will either all be on the same line or will have one line each.
+BinPackArguments: true
+BinPackParameters: true
+
+# The break should always be after the binary and ternary operands.
+BreakBeforeBinaryOperators: None
+BreakBeforeTernaryOperators: true
+
+# Break constructor initializers before the colon and after the commas.
+# Constructor()
+#     : initializer1(),
+#       initializer2()
+BreakConstructorInitializers: BeforeColon
+
+# Pad the braced list with spaces inside the braces.
+Cpp11BracedListStyle: false
+
+# Where to put the & and the * in the case of pointers/references.
+DerivePointerAlignment: false
+PointerAlignment: Left
+
+# Keep empty lines at end of file.
+KeepEmptyLinesAtEOF: false
+
+# Keep empty lines at start of a block.
+KeepEmptyLinesAtTheStartOfBlocks: true
+
+# Align lambda body relative to the lambda signature.
+LambdaBodyIndentation: Signature
+
+# No indent in namespaces.
+NamespaceIndentation: None
+
+# No way to specify C vs C++ style comment for namespace closure, so just
+# leave them alone.
+FixNamespaceComments: false
+
+# 1 space before trailing comments
+SpacesBeforeTrailingComments: 1
+
+# Colon spacing should always be '"key" : "value"'
+BitFieldColonSpacing: Both
+
+# Do not break up strings into multi-line strings. This becomes a problem when
+# multi-line strings with no spaces are broken, which upsets the Tup code checker.
```

Comment:
Having this set to 'false' kills one of clang-format useful and time saving features.
How bad is the problem with that Tup?

Reply:
No, its a tool I used to use.